src/mappings/prices.ts

```diff
@@ -35,9 +36,13 @@ export function _createPrice(priceId: string, tokenId: i32, trade: Trade, event:
   // Addd token
   price.token = BigInt.fromI32(tokenId).toString()
 
-  // Price and volume
+  // Price: The price is given in weis of the token per owl, so it needs to be translated into tokens per owl
   let batchExchange = BatchExchange.bind(event.address)
-  price.priceInOwl = batchExchange.currentPrices(tokenId)
+  let token = getTokenById(tokenId)
+  let priceInWeis = batchExchange.currentPrices(tokenId)
+  price.priceInOwl = toPriceInUnits(priceInWeis, token.decimals, getOwlDecimals())
```

Comment:
Let me know if I got this right.

Before this change, the price was saved raw, which was the price in wei for a given token, based off OWL decimals (18).
Now, the price will be given in token units, converted from OWL (18 decimals) to token (X decimals).

Is that right?

Reply:
* Yes, imagine 2 tokens, one with 1 decimal, one with 2 decimals.
* Volume of token 1 (decimal = 1) = `50`   ---> 5
* Volume of token 2 (decimals =2) = `500`    ----> 5
* The price is `1`
* The contract would give us 500/50 ----> `10`
* This PR takes this `10` and divide it by `10 ** (2 - 1)`

Reply:
Hmmm I see. I'll have to update TheGraphApi to accept the price already in units then.

On a different note, I don't think BigInt can handle decimals -> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt

Reply:
Actually, why here we are working with BigInt and on other places we use a combination of BN and BigNumber?

Reply:
This is not Javascript, I wish :) 

This is assembly script. No BN nor BigNumber.

In other places (JS) we use BN if it's wei (no decimals), or BigNumber (specially for prices with decimals)

Yes, I know there's no decimals in BigInt, this is why the price is expressed in a BigDecimal with precision 18

Reply:
Documenting what we've been discussing offline:

This calculation is correct. What is returned is the priceInOwl in Wei, so a BigInt is enough to represent the value.

Taking as example the current contract value for USDC price (6 decimals of precision): `980942729992317873036301363769`
After going through this function, 12 (18 from OWL - 6 from USDC) last digits are dropped.
The final value is `980942729992317873`.